File: board/src/core/api/trace.ts
import axios from 'axios';
import { ITrace } from "~/core/models";

export const all =  async () => {
  const res = await axios.get('api/trace/all');
  const resData = res.data as string[]
  return resData;
}

export const rangeBy =  async (traceId:string, fromDate:string, toDate:string) => {
  const res = await axios.get('api/trace/range-by', {
    id: traceId,
    from_date:fromDate,
    to_date:toDate
  });
  const resData = res.data as {
    value: number;
    ts: string
  }[];
  return resData 
};

File: app/mlboard/api/services/trace.py
from mlboard.core import models as ms
from mlboard.core import queries as qs
from fastapi import APIRouter, Query
import typing as t
from logging import getLogger
from cytoolz.curried import pipe, map
from pydantic import BaseModel
import uuid
import datetime
from logging import getLogger
logger = getLogger("api.trace")

router = APIRouter()


@router.get('/trace/range-by')
async def search_range(
    tag: str,
    from_date: datetime.datetime,
    to_date: datetime.datetime,
    limit: int = 10000,
) -> t.List[ms.TracePoint]:
    async with qs.get_conn() as conn:
        rows = await qs.TracePoint(conn).range_by(
            tag=tag,
            from_date=from_date,
            to_date=to_date,
            limit=limit
        )
    return rows
